contracts/xasset-orchestrator/src/orchestrator.rs

```diff
@@ -0,0 +1,161 @@
+    fn set_asset_contract(
+        &mut self,
+        asset_symbol: String,
+        asset_contract: Address,
+    ) -> Result<(), Error>;
+    // Manually set an existing asset symbol to an existing contract address. Dangerous!
+    // This should only be used when needing to update an existing symbol's contract.
+    fn set_existing_asset_contract(
+        &mut self,
+        asset_symbol: String,
+        asset_contract: Address,
+    ) -> Result<(), Error>;
+}
+
+impl IsOrchestratorTrait for Storage {
+    fn init(&mut self, xlm_sac: Address, xlm_contract: Address) -> Result<(), Error> {
+        Contract::require_auth();
+        self.xlm_sac.set(&xlm_sac);
+        self.xlm_contract.set(&xlm_contract);
+        Ok(())
+    }
+
+    fn deploy_asset_contract(
+        &mut self,
+        asset_contract: Address,
+        pegged_asset: Symbol,
+        min_collat_ratio: u32,
+        name: String,
+        symbol: String,
+        decimals: u32,
+        annual_interest_rate: u32,
+    ) -> Result<Address, Error> {
+        let env = env();
+        Contract::require_auth();
+        // Check if the asset contract is already deployed
+        if self.assets.has(symbol.clone()) {
+            return Err(Error::AssetAlreadyDeployed);
+        }
+
+        // Upload the xasset wasm to prep for deploy
+        let wasm_hash = env.deployer().upload_contract_wasm(xasset::WASM);
+
+        // Deploy the contract, salting with the symbol
+        let deployed_contract = create_contract(env, &wasm_hash, symbol.clone());
+
+        let contract_address = deployed_contract.map_err(|_| Error::InitFailed)?;
+        // Create a client instance for this contract so we can initialize it
+        let client = xasset::Client::new(env, &contract_address);
+
+        // Initialize the contract
+        client.cdp_init(
+            &self.xlm_sac.get().unwrap(),
+            &&self.xlm_contract.get().unwrap(),
+            &asset_contract,
+            &pegged_asset,
+            &min_collat_ratio,
+            &name,
+            &symbol,
+            &decimals,
+            &annual_interest_rate,
+        );
+
+        // Store the deployed contract address in the assets map
+        self.assets.set(symbol.clone(), &contract_address);
+
+        Ok(contract_address)
+    }
+
+    fn get_asset_contract(&self, asset_symbol: String) -> Result<Address, Error> {
+        if !self.assets.has(asset_symbol.clone()) {
+            return Err(Error::NoSuchAsset);
+        }
+        return Ok(self.assets.get(asset_symbol.clone()).unwrap());
+    }
+
+    fn set_asset_contract(
+        &mut self,
+        asset_symbol: String,
+        asset_contract: Address,
+    ) -> Result<(), Error> {
+        Contract::require_auth();
+        if self.assets.has(asset_symbol.clone()) {
+            return Err(Error::AssetAlreadyDeployed);
+        }
+        self.assets.set(asset_symbol.clone(), &asset_contract);
```

Comment:
I'm not a fan of their use of clone (it's cheaper because it's a reference copy, but still annoying to read and not your fault). Here it is not needed though since we can give up ownership.
```suggestion
        self.assets.set(asset_symbol, &asset_contract);
```